src/Application/Documents/Commands/UpdateDocument.cs

```diff
@@ -1,15 +1,84 @@
+using Application.Common.Exceptions;
+using Application.Common.Interfaces;
 using Application.Common.Models.Dtos.Physical;
+using AutoMapper;
+using FluentValidation;
 using MediatR;
+using Microsoft.EntityFrameworkCore;
 
 namespace Application.Documents.Commands;
 
 public class UpdateDocument
 {
+    public class Validator : AbstractValidator<Command>
+    {
+        public Validator()
+        {
+            RuleLevelCascadeMode = CascadeMode.Stop;
+
+            RuleFor(x => x.Title)
+                .NotEmpty().WithMessage("Title is required.")
+                .MaximumLength(64).WithMessage("Title cannot exceed 64 characters.");
+
+            RuleFor(x => x.Description)
+                .MaximumLength(256).WithMessage("Description cannot exceed 256 characters.");
+
+            RuleFor(x => x.DocumentType)
+                .NotEmpty().WithMessage("DocumentType is required.")
+                .MaximumLength(64).WithMessage("DocumentType cannot exceed 64 characters.");
+        }
+    }
+
     public record Command : IRequest<DocumentDto>
     {
         public Guid DocumentId { get; init; }
         public string Title { get; init; } = null!; 
         public string? Description { get; init; }
         public string DocumentType { get; init; } = null!; 
     }
+    
+    public class CommandHandler : IRequestHandler<Command, DocumentDto>
+    {
+        private readonly IApplicationDbContext _context;
+        private readonly IMapper _mapper;
+
+        public CommandHandler(IApplicationDbContext context, IMapper mapper)
+        {
+            _context = context;
+            _mapper = mapper;
+        }
+
+        public async Task<DocumentDto> Handle(Command request, CancellationToken cancellationToken)
+        {
+            var document = await _context.Documents
+                .Include( x => x.Importer)
+                .FirstOrDefaultAsync( x => x.Id.Equals(request.DocumentId), cancellationToken);
+
+            if (document is null)
+            {
+                throw new KeyNotFoundException("Document does not exist.");
+            }
+
+            var titleExisted = await _context.Documents
+                .Include(x => x.Importer)
+                .AnyAsync(x => 
+                x.Title.Trim().ToLower().Equals(request.Title.Trim().ToLower())
+                && x.Id != document.Id
+                && x.Importer!.Id == document.Importer!.Id
```

Comment:
I actually don't know what to do with this nullable property.

Reply:
I need to confirm that is it in your desire to handle this nullable property like throwing an exception or sth sth ?

Reply:
(x.Importer?.Id ?? throw new NullReferenceException("Importer is null")) == (document.Importer?.Id ?? throw new NullReferenceException("Document's Importer is null")

Reply:
I don't think throwing an exception insice a lamda function is a good idea...
Or is it just me? Or C# problem???